src/Application/Characters/Models/CharacterCompetitiveRatingViewModel.cs

```diff
@@ -0,0 +1,25 @@
+﻿using AutoMapper;
+using Crpg.Application.Common.Interfaces;
+using Crpg.Application.Common.Mappings;
+using Crpg.Application.Common.Services;
+using Crpg.Application.Parties.Models;
+using Crpg.Domain.Entities.Characters;
+using Crpg.Domain.Entities.Parties;
+
+namespace Crpg.Application.Characters.Models;
+
+public record CharacterCompetitiveRatingViewModel : IMapFrom<CharacterRating>
+{
+    public float CompetitiveRating { get; init; }
+    public void Mapping(Profile profile)
+    {
+        profile.CreateMap<Character, CharacterCompetitiveRatingViewModel>()
+            .ForMember(u => u.CompetitiveRating, opt => opt.MapFrom(c => ComputeCompetitiveRating(c.Rating)));
+
+    }
+
+    private double ComputeCompetitiveRating(CharacterRating rating)
+    {
+        return 0.03 * Math.Pow(0.01 * rating.Value - 2 * rating.Deviation, 3.98);
```

Comment:
Not a big fan of this solution. We should not hide logic inside a custom object mapping.

What I suggest instead is to add a column CompetitiveRating to the character that would be updated whenever the rating is updated. Then you could have a `characters/{id}/rating` endpoint that loads all rating columns of a character, and a `leaderboard` endpoint that won't need anymore to load all the characters everytime.

Reply:
This pose a problem as whenever i update the competitive rating formula , we would have old competitive rating mixed with new competitive rating.
Would a task executed every 10 minute to update the leaderboard solve the problem? 
Also, would a service called IComptetiveRatingModel be better to not have logic in a custom object mapping?

Reply:
That would work too!